x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/Cast.java

```diff
@@ -74,6 +78,18 @@ protected Processor makeProcessor() {
         return new CastProcessor(DataTypeConversion.conversionFor(from(), to()));
     }
 
+    @Override
+    public ScriptTemplate asScript() {
+        ScriptTemplate fieldAsScript = asScript(field());
```

Comment:
Since the solution is to use a slightly different script, simply override `processScript` (see its implementations for examples like `MathFunction`). This way you only need to define the string and not worry about the boiler-plate around passing the script further.

Reply:
With `processScript` we cannot set the dataType as parameter and we are forced to send it as String inline. Will leave it as is at the moment, as discussed.